### DISCUSSION TITLE
Recommended Package Tags

### DISCUSSION THREAD

**Discussion by iamcam** [SwiftPackageIndex/SwiftPackageIndex-Server#1359] [Ideas]
I just came across an interesting idea in the Flutter community - "[Flutter Favorite program](https://flutter.dev/docs/development/packages-and-plugins/favorites)". The idea is to provide a curated list of high-quality packages that should be at the top of peoples' lists when building their apps. Each package meets certain criteria, then passes through a committee made of community members.. Once in, the package gets tagged accordingly at [pub.dev](https://pub.dev/flutter/favorites). There are a few perks and flair for project web pages, and membership isn't lifetime - packages can be removed.

I wonder if this might be something worth considering for the Swift Package Index, especially for discoverability and those who are newer to swift development. 

I recognize that this sort of thing can be difficult from so many different angles, so I don't want to try minimizing the logistics and people-issues that can arise, but I think we can try giving the community a fair shake.

**Comment by daveverwer** [comment]
I like this idea, and I think it'd fit very nicely into some of the things we're thinking about with package collections.

As you say, the big thing here would be forming a panel of people who would be responsible for making decisions on what packages would go into any "blessed" list. This is not a job for one or two people. It would need a fairly large set of people with different views and opinions to all be involved to make this worthwhile.

**Comment by iamcam** [comment]
Perhaps running a limited test phase where the team compiles a number of general use packages would be a good start - _What packages are my go-to on any new project?_ and go from there (even if it's a static list). Additionally, it might be worth thinking about tagging projects in some way (perhaps as part of this initiative) so search results can indicate what broad sorts of categories various packages fit into.

**Comment by daveverwer** [comment]
I'm afraid I have to disagree here.

The **most important** thing about this feature, if it ever happens is that the packages are not picked by a small team. If you were suggesting that the SPI team would be the ones responsible for an initial go at this, I'd have to disagree even more as the primary Swift that both @finestructure and I write is server-side, which is not representative of most people using the SPI.

In my opinion, if this happens, it *must* be very much a community effort, including people from many backgrounds and from many different types of development (large companies, as well as small independents).

I'm also thinking about it differently from you, I think. I don't have a list of "go-to" packages, and I think encouraging people to think like installing dependencies as step one is counterproductive. The way I saw what you were thinking was a manual "seal of approval" as a good quality and well-maintained package. We can do some of that with automated metrics, but ultimately a human is better at it. From that list, a set of quality packages would start to emerge.

Tagging/keywords is coming, more details here: https://blog.swiftpackageindex.com/posts/whats-next-may-2021/

**Comment by finestructure** [comment]
As Dave said, which I think can also be summed up with my list of go-to packages for any new project:

```
[]
```

😅

**Comment by iamcam** [comment]
Yep - I totally understand your point @daveverwer - I think maybe my explanation wasn't quite clear enough 😊. Maybe what I meant to say was thinking of this from an MVP standpoint - start small, do enough work to sufficiently test the idea. A small committee (not the SPI team - I like the idea of keeping it separate) could be part of determining the initial set of requirements and packages that fit those reqs.

I'm also on the same page with choice packages -  I personally don't add any packages to projects unless I absolutely need to, but I know there are developers out there who approach development differently and have a few they just love to use on everything. Still - not something I allow on my teams. Speaking of which, I'm coming from the large agency world (well, large company, small mobile teams), so I get it - my use and selection criteria can be quite unique.

So, bringing this full-circle - _yes to all your points_. Let's figure out what getting a seal of approval means (I do like how the Flutter Ecosystem Committee is very clear about their process). Deciding scope for an initial push will be important (I do agree with all-manual up front, but what guardrails should be used to keep scope contained?). What kinds of representation on the committee is desired? etc.

I'm more than happy to lend input if that's the direction you want to go, but perhaps you want to noodle on this a bit first. Either way 👍

**Comment by daveverwer** [comment]
Yes, I agree it would need well-defined and publicly documented criteria. However, I think it does need more thinking about too. It wasn't on our short term plan, and while that doesn't preclude it from happening, it's not something we'd want to rush into.

Is there a way I can drop you a private message? Email or something would be great. If you want to email me, it's my first name at the domain I use for my newsletter.
